OpenRA.Game/Graphics/Animation.cs

```diff
@@ -63,6 +64,7 @@ public IEnumerable<IRenderable> Render(WPos pos, WVec offset, int zOffset, Palet
 				return new IRenderable[] { shadowRenderable, imageRenderable };
 			}
 
+			rendered = true;
```

Comment:
You need to move this to the top of the function. If there is a shadow, the return above will mean you don't set this in time.